zap/src/main/java/org/zaproxy/zap/extension/api/CoreAPI.java

```diff
@@ -876,12 +883,49 @@ public void run() {
             JSONObject newParams = new JSONObject();
             params.put("use", false);
             return getNetworkImplementor().handleApiAction("setUseClientCertificate", newParams);
+        } else if (ACTION_FILE_UPLOAD.equals(name)) {
+            OptionsParamApi apiParam = Model.getSingleton().getOptionsParam().getApiParam();
+
+            if (!apiParam.isFileTransferAllowed()) {
+                throw new ApiException(ApiException.Type.BAD_OTHER, "File transfer not enabled");
+            }
+            Path filePath =
+                    getChildPath(apiParam.getTransferDir(), getParam(params, PARAM_FILENAME, ""));
+
+            try {
+                Files.createDirectories(filePath.getParent());
+                Files.writeString(filePath, getParam(params, PARAM_CONTENTS, ""));
+                LOGGER.debug("Uploaded file to {}", filePath);
+            } catch (IOException e) {
+                LOGGER.error("Failed to create file {}", filePath, e);
+                throw new ApiException(ApiException.Type.ILLEGAL_PARAMETER, PARAM_FILENAME, e);
+            }
```

Comment:
Would it be helpful to return the path that was written to?

Reply:
Instead of just the ok at 906

Reply:
Done